### ISSUE TITLE
version 0.4.1 causes NoMethod Error  undefined method `error' for nil:NilClass

### ISSUE THREAD

**Issue by ccasabona** [rails/importmap-rails#17]
I have been successfully learning turbo-rails gem with importmap-rails and Rails 7 alpha.  Update of import-map from 0.3.4 to 0.4.1 causes the following error when trying to start the rails server:

```
/Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/importmap-rails-0.4.1/lib/importmap/map.rb:16:in `rescue in draw': undefined method `error' for nil:NilClass (NoMethodError)
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/importmap-rails-0.4.1/lib/importmap/map.rb:13:in `draw'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/importmap-rails-0.4.1/lib/importmap/engine.rb:5:in `<class:Engine>'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/importmap-rails-0.4.1/lib/importmap/engine.rb:4:in `<module:Importmap>'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/importmap-rails-0.4.1/lib/importmap/engine.rb:3:in `<main>'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/zeitwerk-2.5.0.beta2/lib/zeitwerk/kernel.rb:35:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/importmap-rails-0.4.1/lib/importmap-rails.rb:6:in `<main>'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/zeitwerk-2.5.0.beta2/lib/zeitwerk/kernel.rb:35:in `require'
	from /Users/charlie/.rvm/rubies/ruby-3.0.2/lib/ruby/site_ruby/3.0.0/bundler/runtime.rb:60:in `block (2 levels) in require'
	from /Users/charlie/.rvm/rubies/ruby-3.0.2/lib/ruby/site_ruby/3.0.0/bundler/runtime.rb:55:in `each'
	from /Users/charlie/.rvm/rubies/ruby-3.0.2/lib/ruby/site_ruby/3.0.0/bundler/runtime.rb:55:in `block in require'
	from /Users/charlie/.rvm/rubies/ruby-3.0.2/lib/ruby/site_ruby/3.0.0/bundler/runtime.rb:44:in `each'
	from /Users/charlie/.rvm/rubies/ruby-3.0.2/lib/ruby/site_ruby/3.0.0/bundler/runtime.rb:44:in `require'
	from /Users/charlie/.rvm/rubies/ruby-3.0.2/lib/ruby/site_ruby/3.0.0/bundler.rb:174:in `require'
	from /Users/charlie/sites/defluris_engine/defluris/config/application.rb:12:in `<main>'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/zeitwerk-2.5.0.beta2/lib/zeitwerk/kernel.rb:35:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/bundler/gems/rails-b4ccee7a8a57/railties/lib/rails/commands/server/server_command.rb:137:in `block in perform'
	from <internal:kernel>:90:in `tap'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/bundler/gems/rails-b4ccee7a8a57/railties/lib/rails/commands/server/server_command.rb:134:in `perform'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/thor-1.1.0/lib/thor/command.rb:27:in `run'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/thor-1.1.0/lib/thor/invocation.rb:127:in `invoke_command'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/thor-1.1.0/lib/thor.rb:392:in `dispatch'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/bundler/gems/rails-b4ccee7a8a57/railties/lib/rails/command/base.rb:87:in `perform'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/bundler/gems/rails-b4ccee7a8a57/railties/lib/rails/command.rb:48:in `invoke'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/bundler/gems/rails-b4ccee7a8a57/railties/lib/rails/commands.rb:18:in `<main>'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	from bin/rails:9:in `<main>'
/Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/importmap-rails-0.4.1/lib/importmap/map.rb:14:in `read': No such file or directory @ rb_sysopen - config/importmap.rb (Errno::ENOENT)
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/importmap-rails-0.4.1/lib/importmap/map.rb:14:in `draw'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/importmap-rails-0.4.1/lib/importmap/engine.rb:5:in `<class:Engine>'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/importmap-rails-0.4.1/lib/importmap/engine.rb:4:in `<module:Importmap>'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/importmap-rails-0.4.1/lib/importmap/engine.rb:3:in `<main>'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/zeitwerk-2.5.0.beta2/lib/zeitwerk/kernel.rb:35:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/importmap-rails-0.4.1/lib/importmap-rails.rb:6:in `<main>'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/zeitwerk-2.5.0.beta2/lib/zeitwerk/kernel.rb:35:in `require'
	from /Users/charlie/.rvm/rubies/ruby-3.0.2/lib/ruby/site_ruby/3.0.0/bundler/runtime.rb:60:in `block (2 levels) in require'
	from /Users/charlie/.rvm/rubies/ruby-3.0.2/lib/ruby/site_ruby/3.0.0/bundler/runtime.rb:55:in `each'
	from /Users/charlie/.rvm/rubies/ruby-3.0.2/lib/ruby/site_ruby/3.0.0/bundler/runtime.rb:55:in `block in require'
	from /Users/charlie/.rvm/rubies/ruby-3.0.2/lib/ruby/site_ruby/3.0.0/bundler/runtime.rb:44:in `each'
	from /Users/charlie/.rvm/rubies/ruby-3.0.2/lib/ruby/site_ruby/3.0.0/bundler/runtime.rb:44:in `require'
	from /Users/charlie/.rvm/rubies/ruby-3.0.2/lib/ruby/site_ruby/3.0.0/bundler.rb:174:in `require'
	from /Users/charlie/sites/defluris_engine/defluris/config/application.rb:12:in `<main>'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/zeitwerk-2.5.0.beta2/lib/zeitwerk/kernel.rb:35:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/bundler/gems/rails-b4ccee7a8a57/railties/lib/rails/commands/server/server_command.rb:137:in `block in perform'
	from <internal:kernel>:90:in `tap'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/bundler/gems/rails-b4ccee7a8a57/railties/lib/rails/commands/server/server_command.rb:134:in `perform'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/thor-1.1.0/lib/thor/command.rb:27:in `run'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/thor-1.1.0/lib/thor/invocation.rb:127:in `invoke_command'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/thor-1.1.0/lib/thor.rb:392:in `dispatch'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/bundler/gems/rails-b4ccee7a8a57/railties/lib/rails/command/base.rb:87:in `perform'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/bundler/gems/rails-b4ccee7a8a57/railties/lib/rails/command.rb:48:in `invoke'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/bundler/gems/rails-b4ccee7a8a57/railties/lib/rails/commands.rb:18:in `<main>'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/bootsnap-1.8.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	from bin/rails:9:in `<main>'

```
If I go back to importmap-rails 0.3.4, the rails server loads properly.

**Comment by websebdev** [comment]
It's because the importmap config has changed.
The default config generated with `rails new` is now like that:
```rb
# config/importmap.rb

# Use Action Cable channels (remember to import "@rails/actionable" in your application.js)
# pin "@rails/actioncable", to: "actioncable.esm.js"
# pin_all_from "app/javascript/channels", under: "channels"

# Use direct uploads for Active Storage (remember to import "@rails/activestorage" in your application.js)
# pin "@rails/activestorage", to: "activestorage.esm.js"

# Use node modules from a JavaScript CDN by running ./bin/importmap

pin "application"
pin "@hotwired/turbo-rails", to: "turbo.js"
pin "@hotwired/stimulus", to: "stimulus.js"
pin "@hotwired/stimulus-importmap-autoloader", to: "stimulus-importmap-autoloader.js"
pin_all_from "app/javascript/controllers", under: "controllers"
```

As you can see, it's no longer nested in`Rails.application.config.importmap.draw do`.

**Comment by dhh** [comment]
Fixed in 0.4.3
